DataStructures/Lists/SLinkedList.cs

```diff
@@ -481,20 +481,22 @@ object System.Collections.IEnumerator.Current
 
             public bool MoveNext()
             {
-                _current = _current.Next;
-
-                return (this._current != null);
+                if (_current != null)
```

Comment:
I think this “if statement” fixes the bug.

why did you change the ‘_doublyLinkedList’ to ‘_list’

Reply:
Because it is not a doubly-linked list (it's singke-linked). I presumed it was a copy-paste error, and it would have complicated things in a separate PR.

Reply:
Yeah… that makes sense.
I think the fix is valid.

well done 👍🏾